File: android/letsfly/app/src/main/java/com/testask/letsfly/util/Loggable.java
package com.testask.letsfly.util;

import android.util.Log;

import java.util.Objects;

/**
 * Created by dbudyak on 24.03.17.
 */

public interface Loggable {
    String DEBUG = "DEBUG";

    default void log(Object o) {
        Log.d(DEBUG, Objects.toString(o));
    }

    default void log(String msg) {
        Log.d(DEBUG, msg);
    }

    default void log(Throwable t) {
        Log.d(DEBUG, t.getLocalizedMessage());
    }
}

File: medbox-android/medbox/src/main/kotlin/ru/medbox/utils/Loggable.kt
package ru.medbox.utils

import android.util.Log
import ru.medbox.BuildConfig

interface Loggable {

    fun log(klass: Any, message: String?) {
        if (BuildConfig.DEBUG) {
            message.let { Log.d("DEBUG={$klass}", message) }
        }
    }

}